packages/components/src/components/checkbox/checkbox.tsx

```diff
@@ -77,7 +76,7 @@ export class Checkbox {
   @Prop({ mutable: true }) public checked?: boolean = false;
 
   /** The id of a form element the checkbox should be associated with. */
-  @Prop() public form?: string;
+  @Prop({ reflect: true }) public form?: string; // The ElementInternals API automatically detects the form attribute
```

Comment:
Maybe we should add React E2E tests to assure that this is working

Reply:
Isn't this already covered by `should include name & value in FormData submit if outside of form`